src/Refactoring-UI/RePushUpMethodDriver.class.st

```diff
@@ -0,0 +1,241 @@
+	^  refactoring privateTransform
+
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> configureAndRunRefactoring [
+
+	self configureRefactoring.
+
+	refactoring failedApplicabilityPreconditions ifNotEmpty: [
+		^ self alertFailedPreconditions ].
+	
+	self setBreakingChangesPreconditions.
+	refactoring failedBreakingChangePreconditions
+		ifEmpty: [ self applyChanges ]
+		ifNotEmpty: [ self handleBreakingChanges ]
+]
+
+{ #category : 'resources' }
+RePushUpMethodDriver >> configureRefactoring [
+
+	refactoring := RBPullUpMethodRefactoring
+		               model: model
+		               pullUp: (methods collect: [ :each | each selector ])
+		               from: class name
+		               to: superclass name
+]
+
+{ #category : 'ui - dialogs' }
+RePushUpMethodDriver >> failedPreconditionsErrorString [
+
+	^ String streamContents: [ :stream |
+		  refactoring failedApplicabilityPreconditions do: [ :cond |
+			  cond check ifFalse: [
+				  cond violationMessageOn: stream.
+				  stream cr ] ] ]
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> gatherUserInput [
+	self selectMethodsAndClass
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> handleBreakingChanges [
+
+	| presenter changes |
+	
+	"Display previous transformations that will fix the fixable breaking conditions"
+	changes := self breakingChoices.
+	
+	
+	presenter := self application newSelectMultiple.
+
+	presenter
+		title: 'There are potential breaking changes';
+		label: self labelBasedOnBreakingChanges;
+		items: changes;
+		display: [ :each | each description ];
+		onAccept: [ :dialog |
+			| selectedChanges |
+			selectedChanges := dialog presenter selectedItems.
+			selectedChanges do: #action.
+			self applyChanges ];
+		openDialog
+]
+
+{ #category : 'ui - dialogs' }
+RePushUpMethodDriver >> labelBasedOnBreakingChanges [
+
+	^ String streamContents: [ :stream |
+		  refactoring failedBreakingChangePreconditions do: [ :cond |
+			  cond check ifFalse: [
+				  cond violationMessageOn: stream.
+				  stream cr ] ].
+		  stream nextPutAll: 'Select the strategies to apply' ]
+]
+
+{ #category : 'for mocks' }
+RePushUpMethodDriver >> methodsAndSuperclassSelectionPresenterClass [
+
+	^ methodSelectionPresenterClass ifNil: [ methodSelectionPresenterClass := StSelectClassAndMethodsPresenter ]
+]
+
+{ #category : 'for mocks' }
+RePushUpMethodDriver >> methodsSelectionPresenterClass: aClass [ 
+
+	methodSelectionPresenterClass := aClass
+]
+
+{ #category : 'initialization' }
+RePushUpMethodDriver >> model: aRBNamespace scopes: refactoringScopes pushUpMethods: methodsList [ 
+	
+	model := aRBNamespace.
+	scopes := refactoringScopes.
+	methods := methodsList.
+	class := methods first origin
+]
+
+{ #category : 'action' }
+RePushUpMethodDriver >> pushUpReferencedInstVars [
+
+	notInstVarRefs violators do: [ :violator | 
+		"We add the pushUpVariable transformation to the refactoring previous transformations"
+		refactoring pushUpVariable: (violator at: 2) ]
+]
```

Comment:
Would it make sense here to encapsulate this `violator at: 2`? Something like `notInstVarRefs referencedInstanceVariables do: [ :iv | refactoring pushUpVariable: iv ]`

Reply:
Yes, I found this code ugly too, I have implemented your suggestion